### PR TITLE
vpa configurable via values.yaml

### PR DESCRIPTION
**Pull request by dergeberl**
**How to categorize this PR?**
/area auto-scaling
/kind enhancement
/priority normal
/os ubuntu

**What this PR does / why we need it**
Allow configuration of the VPA settings in the `values.yaml`. To set the `updateMode` and the `minAllowed`.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Release note**:
```improvement operator
VPA is now configurable via values.yaml in helm chart.
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @dergeberl for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/gardener-extension-os-ubuntu/values.yaml [hunk 1]
```diff
@@ -5,6 +5,15 @@ image:
 
 resources: {}
 
+vpa:
```
Comment (ialidzhikov) [inline charts/gardener-extension-os-ubuntu/values.yaml:8]: You need to run `make generate` to have this change reflected also in `example/controller-registration.yaml` (as this is a change under `charts/gardener-extension-os-ubuntu/`).

Comment (dergeberl) [reply charts/gardener-extension-os-ubuntu/values.yaml:8]: i updated the `example/controller-registration.yaml`

**Review by rfranzke — APPROVED** [review]
/lgtm
/reviewed ok-to-test
/squash

**Review by vlvasilev — APPROVED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
@dergeberl , can you rebase against the master as there is an issue with `make verify` that is fixed in the master?

File: charts/gardener-extension-os-ubuntu/templates/vpa.yaml [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+  name: {{ include "name" . }}-vpa
   namespace: {{ .Release.Namespace }}
 spec:
+  {{- if .Values.vpa.resourcePolicy }}
+  resourcePolicy:
+    containerPolicies:
+      - containerName: '*'
+        minAllowed:
+          cpu: {{ required ".Values.vpa.resourcePolicy.minAllowed.cpu is required" .Values.vpa.resourcePolicy.minAllowed.cpu }}
+          memory: {{ required ".Values.vpa.resourcePolicy.minAllowed.memory is required" .Values.vpa.resourcePolicy.minAllowed.memory }}
```
Comment (vpnachev) [inline charts/gardener-extension-os-ubuntu/templates/vpa.yaml:14]:
```suggestion
          cpu: {{ required "vpa.resourcePolicy.minAllowed.cpu is required" .Values.vpa.resourcePolicy.minAllowed.cpu }}
          memory: {{ required "vpa.resourcePolicy.minAllowed.memory is required" .Values.vpa.resourcePolicy.minAllowed.memory }}
```

File: charts/gardener-extension-os-ubuntu/values.yaml [hunk 3]
```diff
@@ -5,6 +5,15 @@ image:
...
+  enabled: true
+# resourcePolicy:
+#   minAllowed:
+#     cpu: 100m
+#     memory: 128Mi
```
Comment (vpnachev) [inline charts/gardener-extension-os-ubuntu/values.yaml:13]: Please, explicitly enable some minAllowed values or disable by default the VPA.

**Review by vpnachev — CHANGES_REQUESTED** [review]

**Comment by ialidzhikov** [comment]
@dergeberl , can you run `make generate` once again?

```
> Generate / Vendor Check
>> make generate
make generate needs to be run:
 M example/controller-registration.yaml
make: *** [Makefile:80: check-generate] Error 1
```

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**